include/sqlpp11/async_query_service.h

```diff
@@ -0,0 +1,224 @@
+		}
+	};
+
+	template<typename Connection_pool, typename Query, typename Lambda>
+	class query_poster
+	{
+	private:
+		using Pool_connection = decltype((*((Connection_pool*)0)).get_connection());
+		using Result = decltype((*((Pool_connection*)0))(Query()));
+
+		template<class F, class...Args>
+		struct is_callable
+		{
+			template<class U> static auto test(U* p) -> decltype((*p)(std::declval<Args>()...), void(), std::true_type());
+			template<class U> static auto test(...) -> decltype(std::false_type());
+
+			typedef decltype(test<F>(0)) type;
+			static constexpr bool value = decltype(test<F>(0))::value;
+		};
+
+	public:
+		query_poster(Connection_pool& p, Query& q, Lambda& cb) : pool(p), query(q), callback(cb) {}
+
+		void operator()()
+		{
+			Pool_connection connection;
+			try
+			{
+				connection = std::move(pool.get_connection());
+			}
+			catch (const sqlpp::exception& e)
+			{
+				impl(sqlpp::error(sqlpp::error::connection_error, e.what()), std::move(connection), Result(), callback);
+			}
+
+			try
+			{
+				impl(sqlpp::error(sqlpp::error::ok), std::move(connection), std::move(connection(query)), callback);
+			}
+			catch (const std::exception& e)
+			{
+				impl(sqlpp::error(sqlpp::error::query_error, e.what()), std::move(connection), Result(), callback);
+			}
+		}
+
+		template<typename Lambda, typename std::enable_if<!is_callable<Lambda>::value &&
+			!is_callable<Lambda, sqlpp::error>::value &&
+			!is_callable<Lambda, sqlpp::error, Result>::value &&
+			!is_callable<Lambda, sqlpp::error, Result, Pool_connection>::value, int>::type = 0>
+		void impl(sqlpp::error error, Pool_connection connection, Result result, Lambda& callback)
+		{
+			static_assert(false, "Callback signature is incompatible. Refer to the wiki for further instructions.");
+		}
+
+		template<typename Lambda, typename std::enable_if<is_callable<Lambda>::value, int>::type = 0>
+		void impl(sqlpp::error error, Pool_connection connection, Result result, Lambda& callback)
+		{
+			callback();
+		}
+
+		template<typename Lambda, typename std::enable_if<is_callable<Lambda, sqlpp::error>::value, int>::type = 0>
+		void impl(sqlpp::error error, Pool_connection connection, Result result, Lambda& callback)
+		{
+			callback(std::move(error));
+		}
+
+		template<typename Lambda, typename std::enable_if<is_callable<Lambda, sqlpp::error, Result>::value, int>::type = 0>
+		void impl(sqlpp::error error, Pool_connection connection, Result result, Lambda& callback)
+		{
+			callback(std::move(error), std::move(result));
+		}
+
+		template<typename Lambda, typename std::enable_if<is_callable<Lambda, sqlpp::error, Result, Pool_connection>::value, int>::type = 0>
+		void impl(sqlpp::error error, Pool_connection connection, Result result, Lambda& callback)
+		{
+			callback(std::move(error), std::move(result), std::move(connection));
+		}
+
+		Connection_pool& pool;
+		Query query;
+		Lambda callback;
+	};
+
+	using namespace std::chrono_literals;
```

Comment:
I removed the whole asio_query_service because the more I worked on it the more useless I felt it is. See previous comment on documentation for my reason to remove async_query_service.h